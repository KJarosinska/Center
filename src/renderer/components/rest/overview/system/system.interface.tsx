export interface SystemProps {
  osVersion: string
  lastUpdate?: string
  onUpdatesCheck?: () => void
  updateAvailable?: boolean
}
