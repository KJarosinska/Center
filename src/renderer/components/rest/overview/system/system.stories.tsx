import { storiesOf } from "@storybook/react"
import React from "react"
import System from "Renderer/components/rest/overview/system/system.component"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import getFakeAdapters from "App/tests/get-fake-adapters"
import { action } from "@storybook/addon-actions"
import styled from "styled-components"

const fakeSystemInfo = getFakeAdapters().purePhone
const osVersion = fakeSystemInfo.getOsVersion()
const updated = new Date(fakeSystemInfo.getOsUpdateDate()).toLocaleDateString(
  "en-US"
)

const Part = styled.div`
  padding: 2rem;
  > p {
    margin-bottom: 2rem;
  }
`

storiesOf("Overview/System", module).add("System", () => {
  return (
    <div style={{ maxWidth: "59rem" }}>
      <Part>
        <Text displayStyle={TextDisplayStyle.SmallText}>Not updated yet</Text>
        <System
          osVersion={osVersion}
          onUpdateCheck={action("checked for update")}
        />
      </Part>
      <Part>
        <Text displayStyle={TextDisplayStyle.SmallText}>
          Updated some time ago
        </Text>
        <System
          osVersion={osVersion}
          lastUpdate={updated}
          onUpdateCheck={action("checked for update")}
        />
      </Part>
      <Part>
        <Text displayStyle={TextDisplayStyle.SmallText}>
          New update available
        </Text>
        <System
          osVersion={osVersion}
          lastUpdate={updated}
          onUpdate={action("update requested")}
          updateAvailable
        />
      </Part>
      <Part>
        <Text displayStyle={TextDisplayStyle.SmallText}>
          New (first) update available
        </Text>
        <System
          osVersion={osVersion}
          onUpdate={action("update requested")}
          updateAvailable
        />
      </Part>
      <Part>
        <Text displayStyle={TextDisplayStyle.SmallText}>
          Updated just now (custom text)
        </Text>
        <System
          osVersion={osVersion}
          lastUpdate={"just now"}
          onUpdateCheck={action("checked for update")}
        />
      </Part>
    </div>
  )
})
