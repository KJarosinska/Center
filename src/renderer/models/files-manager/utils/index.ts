import { FilesManagerState } from "Renderer/models/files-manager/files-manager.interface"

export const prepareDataForStackedBarChart = (data: FilesManagerState) => {
  return data.memoryData.map(el => {
    const { filesType, filesAmount, occupiedMemory, ...rest } = el
    return {
      value: occupiedMemory,
      ...rest,
    }
  })
}
