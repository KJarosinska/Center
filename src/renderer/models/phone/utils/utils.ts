import Faker from "faker"

// TODO: remove before production
export const generateFakeData = (numberOfContacts: number) => {
  return Array(numberOfContacts)
    .fill(0)
    .map(_ => ({
      id: Faker.random.uuid(),
      firstName: Faker.name.firstName(),
      lastName: Faker.name.lastName(),
      phoneNumber: Faker.phone.phoneNumber(),
      favourite: Faker.random.boolean(),
    }))
}

export const generateSortedStructure = (fakeState: any) => {
  const alphabet = new Array(26)
    .fill(1)
    .map((_, i) => String.fromCharCode(65 + i))

  const sortedContactList = fakeState.sort((a: any, b: any) =>
    a.firstName.localeCompare(b.firstName)
  )

  const generateFakeStructure = () => {
    const fakeStructure = []

    for (const letter of alphabet) {
      fakeStructure.push({
        category: letter,
        contacts: [],
      })
    }

    fakeStructure.unshift({
      category: "Favs",
      contacts: [],
    })

    return fakeStructure
  }

  const placeContactsInStructure = () => {
    const fakeStructure: any = generateFakeStructure()
    fakeStructure.forEach((fakeContact: any) => {
      const contactLetter = fakeContact.category
      const contactList: string[] = fakeContact.contacts
      sortedContactList.forEach((sortedContact: any) => {
        if (
          sortedContact.firstName.charAt(0) === contactLetter &&
          !sortedContact.favourite
        ) {
          contactList.push(sortedContact)
        }
      })
    })

    sortedContactList.forEach((contacts: any) => {
      if (contacts.favourite) {
        fakeStructure[0].contacts.push(contacts)
      }
    })

    return fakeStructure
  }

  return placeContactsInStructure()
}

export const sanitizeContacts = (structure: any) => {
  return structure.filter((el: any) => el.contacts.length > 0)
}

export const filterContacts = (contacts: any, substring: string) => {
  return contacts.map((contactsByLetter: any) => ({
    ...contactsByLetter,
    contacts: contactsByLetter.contacts.filter((contact: any) => {
      const filterableFields = Object.values(contact).filter(value => {
        return typeof value === "string"
      })
      return filterableFields.some((value: any) =>
        value.toLowerCase().includes(substring.toLowerCase())
      )
    }),
  }))
}
