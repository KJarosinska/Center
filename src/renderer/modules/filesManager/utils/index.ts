import {
  FilesManagerData,
  FilesManagerState,
} from "Renderer/models/files-manager/files-manager.interface"

export const memoryDataWithConvertedBytes = (data: FilesManagerState) => {
  return data.memoryData.map((element: FilesManagerData) => ({
    ...element,
    occupiedMemory: element.occupiedMemory / Math.pow(1024, 3),
  }))
}
