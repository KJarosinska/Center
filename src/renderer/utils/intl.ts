import { createIntl } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"

export const intl = createIntl({
  locale: LANGUAGE.default,
  messages: localeEn,
})
