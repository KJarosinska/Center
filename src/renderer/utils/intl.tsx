import React from "react"
import { createIntl } from "react-intl"
import { LANGUAGE } from "Renderer/constants/languages"
import localeEn from "Renderer/locales/main/en-US.json"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"

const testLocale = extractLanguageKeys(localeEn)

export const intl = createIntl({
  locale: LANGUAGE.default,
  messages: process.env.NODE_ENV === "test" ? testLocale : localeEn,
})

export const textFormatters = {
  b: (str: string) => <strong>{str}</strong>,
  u: (str: string) => <u>{str}</u>,
  i: (str: string) => <em>{str}</em>,
  s: (str: string) => <s>{str}</s>,
}
