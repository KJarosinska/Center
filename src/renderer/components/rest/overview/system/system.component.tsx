import React from "react"
import FunctionComponent from "Renderer/types/function-component.interface"
import { SystemProps } from "Renderer/components/rest/overview/system/system.interface"
import Card, {
  CardAction,
  CardActionButton,
  CardText,
} from "Renderer/components/rest/overview/card.elements"
import styled from "styled-components"
import Text, {
  TextDisplayStyle,
} from "Renderer/components/core/text/text.component"
import { FormattedMessage } from "react-intl"
import { intl } from "Renderer/utils/intl"
import Reload from "Renderer/svg/circle-arrow.svg"
import {
  fontWeight,
  letterSpacing,
  textColor,
} from "Renderer/styles/theming/theme-getters"
import { noop } from "Renderer/utils/noop"

const TextInfo = styled(CardText)``

const Version = styled.div`
  display: flex;
  align-items: baseline;

  span {
    margin-left: 0.8rem;
    font-weight: ${fontWeight("light")};
    text-transform: none;
  }
`

const LastUpdate = styled(Text)`
  margin-top: 1.2rem;
  letter-spacing: ${letterSpacing("small")}rem;
  color: ${textColor("placeholder")};
`

const AvailableUpdate = styled(Text)`
  margin-top: 1.6rem;
  text-transform: none;
  letter-spacing: ${letterSpacing("small")}rem;
`

const System: FunctionComponent<SystemProps> = ({
  className,
  osVersion,
  lastUpdate,
  updateAvailable,
  onUpdateCheck = noop,
  onUpdate = noop,
}) => {
  const checkUpdates = () => onUpdateCheck()
  const update = () => onUpdate()

  return (
    <Card className={className}>
      <TextInfo>
        <Version>
          <Text displayStyle={TextDisplayStyle.SecondaryBoldHeading}>
            {osVersion}
          </Text>
          <Text element={"span"} displayStyle={TextDisplayStyle.SmallText}>
            <FormattedMessage id="view.name.overview.system.version" />
          </Text>
        </Version>
        {Boolean(lastUpdate) && (
          <LastUpdate displayStyle={TextDisplayStyle.SmallFadedText}>
            <FormattedMessage
              id="view.name.overview.system.lastUpdate"
              values={{ date: lastUpdate }}
            />
          </LastUpdate>
        )}
        {updateAvailable && (
          <AvailableUpdate displayStyle={TextDisplayStyle.SmallText}>
            <FormattedMessage id="view.name.overview.system.updateAvailable" />
          </AvailableUpdate>
        )}
      </TextInfo>
      <CardAction filled>
        {updateAvailable ? (
          <CardActionButton
            active
            label={intl.formatMessage({
              id: "view.name.overview.system.updateAction",
            })}
            Icon={Reload}
            onClick={update}
          />
        ) : (
          <CardActionButton
            active
            label={intl.formatMessage({
              id: "view.name.overview.system.checkForUpdates",
            })}
            Icon={Reload}
            onClick={checkUpdates}
          />
        )}
      </CardAction>
    </Card>
  )
}

export default System
