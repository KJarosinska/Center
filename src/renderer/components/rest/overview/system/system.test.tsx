import "@testing-library/jest-dom/extend-expect"
import React from "react"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import { SystemProps } from "Renderer/components/rest/overview/system/system.interface"
import getFakeAdapters from "App/tests/get-fake-adapters"
import { intl } from "Renderer/utils/intl"
import System from "Renderer/components/rest/overview/system/system.component"
import { fireEvent } from "@testing-library/dom"
import { wait } from "@testing-library/react"

const fakeSystemInfo = getFakeAdapters().purePhone
const lastUpdate = fakeSystemInfo.getOsUpdateDate()
const fakeOsVersion = fakeSystemInfo.getOsVersion()

const renderSystem = ({
  osVersion = fakeOsVersion,
  ...props
}: Partial<SystemProps> = {}) => {
  return renderWithThemeAndIntl(<System osVersion={osVersion} {...props} />)
}

test("matches snapshot", () => {
  const { container } = renderSystem()
  expect(container).toMatchSnapshot()
})

test("renders os version properly", () => {
  const { getByText } = renderSystem()
  expect(getByText(fakeOsVersion)).toBeInTheDocument()
  expect(
    getByText(intl.formatMessage({ id: "view.name.overview.system.version" }))
  ).toBeInTheDocument()
})

test("renders last update info properly", () => {
  const { getByText } = renderSystem({ lastUpdate })
  expect(
    getByText(
      intl.formatMessage(
        { id: "view.name.overview.system.lastUpdate" },
        { date: lastUpdate }
      )
    )
  ).toBeInTheDocument()
})

test("renders available update info properly", () => {
  const { getByText } = renderSystem({ updateAvailable: true })
  expect(
    getByText(
      intl.formatMessage({ id: "view.name.overview.system.updateAvailable" })
    )
  ).toBeInTheDocument()
})

test("renders 'check for updates' button properly", () => {
  const { queryByRole } = renderSystem()
  expect(queryByRole("button")).toHaveTextContent(
    intl.formatMessage({ id: "view.name.overview.system.checkForUpdates" })
  )
})

test("renders 'update now' button properly", () => {
  const { queryByRole } = renderSystem({ updateAvailable: true })
  expect(queryByRole("button")).toHaveTextContent(
    intl.formatMessage({ id: "view.name.overview.system.updateAction" })
  )
})

test("checks for update after button click", async () => {
  const onUpdateCheck = jest.fn()
  const onUpdate = jest.fn()

  const { getByRole } = renderSystem({
    onUpdateCheck,
    onUpdate,
  })

  fireEvent.click(getByRole("button"))

  await wait(() => {
    expect(onUpdateCheck).toHaveBeenCalled()
    expect(onUpdate).not.toHaveBeenCalled()
  })
})

test("triggers update after button click", async () => {
  const onUpdateCheck = jest.fn()
  const onUpdate = jest.fn()

  const { getByRole } = renderSystem({
    updateAvailable: true,
    onUpdateCheck,
    onUpdate,
  })

  fireEvent.click(getByRole("button"))

  await wait(() => {
    expect(onUpdateCheck).not.toHaveBeenCalled()
    expect(onUpdate).toHaveBeenCalled()
  })
})
