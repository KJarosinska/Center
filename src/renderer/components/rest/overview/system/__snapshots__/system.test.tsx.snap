// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c3 {
  margin: 0;
  font-size: 2.4rem;
  line-height: 0.8;
  font-weight: 700;
  -webkit-letter-spacing: -0.06rem;
  -moz-letter-spacing: -0.06rem;
  -ms-letter-spacing: -0.06rem;
  letter-spacing: -0.06rem;
}

.c4 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 1.4rem;
  width: 1.4rem;
  position: relative;
}

.c9 svg {
  width: 100%;
  height: 100%;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c7:hover {
  background: #232426;
}

.c7 g {
  fill: #ffffff;
}

.c10 {
  margin: 0 0.8rem 0 0;
}

.c11 {
  white-space: nowrap;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
}

.c0 {
  display: grid;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  grid-template-areas: "Text Buttons";
  grid-template-columns: auto minmax(18rem,1fr);
  height: 14.4rem;
  padding: 0 4.8rem;
  box-sizing: border-box;
  border-radius: 0.5rem;
  background-color: #ffffff;
}

.c6 {
  grid-area: Buttons;
  justify-self: end;
  min-width: 18rem;
}

.c8 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  z-index: 3;
  padding: 0 1.6rem;
  width: 50%;
}

.c8:disabled {
  pointer-events: none;
}

.c8:hover {
  z-index: 2;
}

.c8:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c8:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c8:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c1 {
  grid-area: Text;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
}

.c2 span {
  margin-left: 0.8rem;
  font-weight: 300;
  text-transform: none;
}

<div>
  <section
    class="c0"
  >
    <div
      class="sc-fznKkj c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          3.1.0
        </h2>
        <span
          class="c4"
        >
          [value] view.name.overview.system.version
        </span>
      </div>
    </div>
    <section
      class="c5 c6"
    >
      <button
        class="c7 sc-fzozJi c8"
        size="2"
        type="button"
      >
        <div
          class="c9 c10"
          data-testid="icon-Reload"
          height="1.4"
          width="1.4"
        >
          <test-file-stub
            image="test-file-stub"
          />
        </div>
        <p
          class="c4 c11"
        >
          [value] view.name.overview.system.checkForUpdates
        </p>
      </button>
    </section>
  </section>
</div>
`;
